src/service.rs

```diff
@@ -143,86 +142,113 @@ impl<Types: NodeType> GlobalState<Types> {
 
         // update the builder state to last built block
         self.builder_state_to_last_built_block
-            .insert(builder_vid_commitment, response_msg);
+            .insert((builder_vid_commitment, view_num), response_msg);
     }
 
     // remove the builder state handles based on the decide event
     pub fn remove_handles(
         &mut self,
         builder_vid_commitment: &VidCommitment,
-        block_hashes: HashSet<(Types::Time, BuilderCommitment)>,
+        block_hashes: HashSet<(VidCommitment, BuilderCommitment, Types::Time)>,
```

Comment:
Why do we need both commitments? These are two different commitments to the same thing (a payload) right?

Reply:
A single builder can build multiple blocks therefore, builder commitment would be different.

Reply:
Note: this is the commitment to what the builder is proposing. We're still (unfortunately) using the same type for the commitment to what the builder offers, and the identifying commitment matching DA Proposal to Quorum Proposal.

Reply:
But the VidCommitment is for view N, the BuilderCommitment is for the proposed block (presumed to have N as its justify_qc view), and the Types::Time should be N...